File: auth-service/src/main/java/com/auth/service/repository/AccountRepository.java
package com.auth.service.repository;

import com.auth.service.model.AccountEntity;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import java.util.Optional;
import java.util.UUID;

@Repository
public interface AccountRepository extends JpaRepository<AccountEntity, UUID> {

	Optional<AccountEntity> findFirstByEmail(String email);

}

File: accounting-service/src/main/kotlin/uberpopug/accounting/repository/AccountRepository.kt
package uberpopug.accounting.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.repository.CrudRepository
import uberpopug.accounting.model.Account
import java.util.*

interface AccountRepository : CrudRepository<Account, Long> {
    fun findAccountByPublicId(publicId: String?): Optional<Account>
    fun findAccountByEmail(email: String?): Optional<Account>
}